csharp/LogicalWrite.cs

```diff
@@ -449,6 +479,85 @@ public static void ConvertDecimal128(ReadOnlySpan<decimal?> source, Span<short>
             }
         }
 
+        public static void ConvertDecimal(ReadOnlySpan<decimal> source, Span<FixedLenByteArray> destination, decimal multiplier, ByteBuffer byteBuffer, int typeLength)
+        {
+            for (int i = 0; i < source.Length; ++i)
+            {
+                var byteArray = byteBuffer.Allocate(typeLength);
```

Comment:
Not sure if it really matters, but I think we can make it faster by allocating a single buffer for all values?

Reply:
The pointer to the byteArray gets stored in the FixedLengthByteArray values that are then read from the C++ library, so we couldn't reuse the same byteArray with length kept as typeLength, but I guess we could allocate a block of byteArray * source.Length which might be a bit faster by reducing the overhead of calls `byteBuffer.Allocate`. Is that what you mean?

We use the same pattern in a few other places, although usually via the `FromFixedLength` method, so I think this should be a separate PR if it does speed things up.

Reply:
> but I guess we could allocate a block of byteArray * source.Length which might be a bit faster by reducing the overhead of calls `byteBuffer.Allocate`. Is that what you mean?

Yes, exactly, this is what I meant. Possibly, the speed improvement is going to be minimal anyway. It is definitely not necessary to amend this PR, it is good as it is now.